Core/HistoryManager.swift

```diff
@@ -85,10 +111,26 @@ public class HistoryManager: HistoryManaging {
         }
     }
 
-    public func loadStore() {
-        historyCoordinator.loadHistory {
-            // Do migrations here if needed
+    public func loadStore(onCleanFinished: @escaping () -> Void) throws {
```

Comment:
Even if the feature is disabled by the user we need to load and clean the database. This needs to happen early on so that the user can flip the setting on and off without a restart.